api/src/main/java/org/openmrs/module/sync/api/impl/SyncServiceImpl.java

```diff
@@ -296,50 +296,59 @@ public List<SyncRecord> getSyncRecords(SyncRecordState[] states, Integer maxSync
 	 * @see org.openmrs.module.sync.api.SyncService#getSyncRecords(org.openmrs.module.sync.SyncRecordState[],
 	 *      org.openmrs.module.sync.server.RemoteServer, java.lang.Integer, java.lang.Integer)
 	 */
-	@Transactional(readOnly = true)

```

Comment:
The read only transaction annotation on here just seems wrong, as this method is actually updating sync records.  I updated the method name just to make it a bit more obvious that it is doing more than fetching records.